arviz/plots/ecdfplot.py

```diff
@@ -0,0 +1,341 @@
+
+    else:
+        if pit:
+            x = np.linspace(1 / npoints, 1, npoints)
+            probs = cdf(values)
+        else:
+            x = np.linspace(values[0], values[-1], npoints)
+            probs = values
+
+        lower, higher = None, None
+        ## This block is for whether to plot ECDF or ECDF-difference
+        if not difference:
+            x_coord, y_coord = get_ecdf_points(x, probs, difference)
+        else:
+            ## Here we subtract the ecdf value as here we are plotting the ECDF-difference
+            x_coord, y_coord = get_ecdf_points(x, probs, difference)
+            for i, x_i in enumerate(x):
+                y_coord[i] = y_coord[i] - (
+                    x_i if pit else cdf(x_i) if cdf else compute_ecdf(values2, x_i)
+                )
+
+    ecdf_plot_args = dict(
+        x_coord=x_coord,
+        y_coord=y_coord,
+        x_bands=x,
+        lower=lower,
+        higher=higher,
+        confidence_bands=confidence_bands,
+        figsize=figsize,
+        fill_band=fill_band,
+        plot_kwargs=plot_kwargs,
+        fill_kwargs=fill_kwargs,
+        plot_outline_kwargs=plot_outline_kwargs,
+        ax=ax,
+        show=show,
+        backend_kwargs=backend_kwargs,
+        **kwargs
+    )
+
+    if backend is None:
+        backend = rcParams["plot.backend"]
+    backend = backend.lower()
+
+    plot = get_plotting_function("plot_ecdf", "ecdfplot", backend)
+    ax = plot(**ecdf_plot_args)
+
+    return ax
+
+
+def compute_ecdf(sample, z):
+    """Compute ECDF.
+
+    This function computes the ecdf value at the evaluation point
+        or a sorted set of evaluation points.
+    """
+    if not isinstance(z, np.ndarray):
+        ## if z is just an instance then use Binary search
+        left, right = 0, len(sample) - 1
+        while left <= right:
+            mid = int((left + right) / 2)
+            if sample[mid] > z:
+                right = mid - 1
+            else:
+                left = mid + 1
+        return left / len(sample)
+    else:
+        ## if z is a list then follow this approach
+        f_z = np.empty(len(z))
+        u_idx = 0
+        for i, z_i in enumerate(z):
+            while u_idx < len(sample) and sample[u_idx] < z_i:
+                u_idx += 1
+            f_z[i] = u_idx + 1
+
+        return f_z / len(sample)
+
+
+def get_ecdf_points(x, probs, difference):
+    """Compute the coordinates for the ecdf points using compute_ecdf."""
+    if difference:
+        x_coord, y_coord = np.empty(len(x)), np.empty(len(x))
+    else:
+        x_coord, y_coord = np.empty(len(x) + 1), np.empty(len(x) + 1)
+        ## pseudo point at the start of ecdf plot so that it touches x-axis
+        x_coord[0], y_coord[0] = x[0], 0
+
+    for i, x_i in enumerate(x):
+        f_x_i = compute_ecdf(probs, x_i)
+        if difference:
+            x_coord[i], y_coord[i] = x_i, f_x_i
+        else:
+            x_coord[i + 1], y_coord[i + 1] = x_i, f_x_i
+    return x_coord, y_coord
```

Comment:
What about something like this for simplicity?

```suggestion
    y = compute_ecdf(probs, x)

    if not difference:
        x = np.insert(x, 0, x[0])
        y = np.insert(y, 0, 0)
    return x, y
```